### ISSUE TITLE
D3D stride issues

### ISSUE THREAD

**Issue by flibitijibibo** [thatcosmonaut/SDL#181]
We've got two that _may_ be related, but we're not sure yet. The cube blit test for #162 is showing an odd overread in D3D12, but we also have this report:

> I did notice an issue though with d3d11, it looks like some stride error regarding textures. Here's some text rendering on dx12/vulkan (no issues here, looks as expected):
> 
> ![image](https://github.com/user-attachments/assets/f98ef185-0268-4bb9-a9bc-5a5bbeee5f72)
> 
> And with no code changes at all, other than switching to dx11, it seems perhaps a stride related issue:
> 
> ![image](https://github.com/user-attachments/assets/27d93446-b36e-458e-b0d4-47a7cca137b0)
> 
> This might be something weird with shader compilation differences on my end, though I'm not sure and haven't debugged it yet. Wanted to mention just in case someone might already have an idea.

@RandyGaul, do you happen to have RenderDoc captures for the three APIs?

**Comment by flibitijibibo** [comment]
The blit thing I believe turned out to be simultaneous target writes, so it's just down to the text rendering sample which we'll need a test program for.

**Comment by RandyGaul** [comment]
With the newest version I had trouble starting up a D3D11 backend. An error saying D3D11 hint is not a valid hint. Are you able to start up a D3D11 backend somehow?

**Comment by flibitijibibo** [comment]
It might be the new CreateDevice signature - does it include DXBC in the bitmask?

**Comment by RandyGaul** [comment]
Just SPIRV for shader formats. D3D11 was specified like this:

```c
// Create the GPU device.
const char* device_name = NULL;
if (use_dx11) device_name = "D3D11";
else if (use_dx12) device_name = "D3D12";
else if (use_metal) device_name = "Metal";
else if (use_vulkan) device_name = "Vulkan";
device = SDL_GpuCreateDevice(SDL_GPU_SHADERFORMAT_SPIRV, true, false, device_name);
```

The error was `ERROR: SDL_HINT_GPU_BACKEND D3D11 unsupported!`.

**Comment by RandyGaul** [comment]
I also tried `SDL_GpuCreateDeviceFromProperties` and found the same error.

**Comment by thatcosmonaut** [comment]
You need to include `SDL_GPU_SHADERFORMAT_DXBC` for D3D11, even if you're using our shadercross tool it's passing DXBC through to D3D11. We made this change because we don't want to surprise clients by selecting backends that they don't have shader support for.

**Comment by RandyGaul** [comment]
I'm not sure I'll be able to produce a minimal repro as it would take a lot of time, but it's in an open source project, easily seen in [the `metaballs` sample](https://github.com/RandyGaul/cute_framework/blob/gpu/samples/metaballs.cpp). I also have renderdoc captures for 3 backends.

[rdc.zip](https://github.com/user-attachments/files/16647730/rdc.zip)

**Comment by flibitijibibo** [comment]
Updating shadercross should give you a new function, SDL_ShaderCross_GetShaderFormats(), that will provide the correct bitflags automatically.

**Comment by thatcosmonaut** [comment]
Resolved by #211 and #212
